point.py

```diff
@@ -0,0 +1,111 @@
+import analytics
+import math
+import random
+import random
+import numpy as np
+
+
+class Point:
+
+    def __init__(self, x = 0, y = 0, mark = ''):
+        self.x = x
+        self.y = y
+        self.mark = mark
+
+    def __add__(self, other):
+        return Point(self.x + other.x, self.y + other.y)
+
+    def __eq__(self, other):
+        return self.x == other.x and self.y == other.y
+
+    def __radd__(self, other):
+        return Point(self.x + other, self.y + other)
+
+    def check_coincident(self, b):
+
+        if b.x == self.x and b.y == self.y:
+            return True
+
+    def shift_point(self, x_shift, y_shift):
+
+        self.x += x_shift
+        self.y += y_shift
+
+class PointPattern(object):
+    
+    def __init__(self):
+        self.points = []
+        self.marks = []
+        self.length = len(self.points)
+
+    def __len__(self):
+        count = 0
+        for item in self.points:
+            count += 1
+        return count
+
+    def add_point(self, point):
+        self.points.append(point)
+
+    def remove_point(self, index):
+        del(self.points[index])
+
+    def average_nearest_neighor(self, mark=None):
+        return analytics.average_nearest_neighbor_distance(self.points, 
+                mark)
+
+    def count_coincident(self):
+        counted = []
+        count = 0
+        for index, point in enumerate(self.points):
```

Comment:
Think about two things: (1) can you use `__eq__` here and (2) is A == B does B == A?